numpyro/distributions/transforms.py

```diff
@@ -1288,6 +1288,98 @@ def __eq__(self, other):
         )
 
 
+class RecursiveLinearTransform(Transform):
+    """
+    Apply a linear transformation recursively such that
+    :math:`y_t = A y_{t - 1} + x_t` for :math:`t > 0`, where :math:`x_t` and :math:`y_t`
+    are vectors and :math:`A` is a transition matrix. The series is initialized by
+    :math:`y_0 = 0`.
+
+    :param transition_matrix: Transition matrix :math:`A` for successive states.
```

Comment:
> Maybe `'matrix'` -> `squared matrix` for clarity.

👍 

> Currently, the bias `x` is time-dependent, but the matrix is constant. Do you plan to make the class name more verbose to reflect that?

Sorry, I explained poorly. The `x` here is the argument of the transform rather than the bias in the `AffineTransform`, for example. Keeping the transition matrix constant means that the transform can be applied to sequences of arbitrary length.

Reply:
x is the (time-dependent) bias (or noise if we place a normal distribution over it) of a linear dynamical model. Typically, people use other notations, something like x_t = Ax{t-1} + b_t.

Reply:
Fair. I was using the x/y notation here to stick with the typical arguments of the `Transform` classes. Do you have an idea for a more explanatory name?

Reply:
The name looks 👌 to me. :)